src/main/app/controller/GetController.ts

```diff
@@ -5,25 +5,32 @@ import { commonContent } from '../../steps/common/common.content';
 
 import { AppRequest } from './AppRequest';
 
+type DivorceOrCivilContext = Record<string, unknown>;
+type TranslationContent = (isDivorce: boolean) => Record<'en' | 'cy' | 'common', DivorceOrCivilContext>;
+
 @autobind
 export class GetController {
-  constructor(protected readonly name: string, protected readonly content: Record<string, Record<string, unknown>>) {}
+  constructor(protected readonly view: string, protected readonly getContent: TranslationContent) {}
 
   public async get(req: AppRequest, res: Response): Promise<void> {
-    if (res.locals.isError) {
+    if (res.locals.isError || res.headersSent) {
       // If there's an async error, it wil have already rendered an error page upstream,
       // so we don't want to call render again
       return;
     }
 
-    const languageContent = this.content[req.session.lang] || this.content['en'] || {};
-    const commonLanguageContent = commonContent[req.session.lang] || commonContent['en'];
-    const commonPageContent = this.content.common || {};
+    const isDivorce = res.locals?.serviceType !== 'civil';
+    const content = this.getContent(isDivorce);
```

Comment:
would this not return an error on the contents which arent high order functions?

Reply:
👍 Nice idea, I've added a switch that will handle both a HOC and a standard object in ad6f948